iis-wf/iis-wf-import/src/main/java/eu/dnetlib/iis/wf/importer/infospace/ImportInformationSpaceJobUtils.java

```diff
@@ -0,0 +1,80 @@
+     * and object store ids will be the same.
+     *
+     * @param dedupRelation              {@link IdentifierMapping} RDD containing mappings from deduplicated ids to original ids.
+     * @param sourceDataset              {@link eu.dnetlib.dhp.schema.oaf.Dataset} RDD.
+     * @param sourceOtherResearchProduct {@link eu.dnetlib.dhp.schema.oaf.OtherResearchProduct} RDD.
+     * @param sourcePublication          {@link eu.dnetlib.dhp.schema.oaf.Publication} RDD.
+     * @param sourceSoftware             {@link eu.dnetlib.dhp.schema.oaf.Software} RDD.
+     * @param spark                      Instance of SparkSession.
+     * @return An RDD of {@link IdentifierMapping} with object store ids.
+     */
+    public static JavaRDD<IdentifierMapping> produceObjectStoreId(JavaRDD<eu.dnetlib.dhp.schema.oaf.Dataset> sourceDataset,
+                                                                  JavaRDD<eu.dnetlib.dhp.schema.oaf.OtherResearchProduct> sourceOtherResearchProduct,
+                                                                  JavaRDD<eu.dnetlib.dhp.schema.oaf.Publication> sourcePublication,
+                                                                  JavaRDD<eu.dnetlib.dhp.schema.oaf.Software> sourceSoftware,
+                                                                  JavaRDD<IdentifierMapping> dedupRelation,
+                                                                  SparkSession spark) {
+        Dataset<eu.dnetlib.dhp.schema.oaf.Dataset> sourceDatasetDS = spark.createDataset(
+                sourceDataset.rdd(), Encoders.bean(eu.dnetlib.dhp.schema.oaf.Dataset.class));
+        Dataset<eu.dnetlib.dhp.schema.oaf.OtherResearchProduct> sourceOtherResearchProductDS = spark.createDataset(
+                sourceOtherResearchProduct.rdd(), Encoders.bean(eu.dnetlib.dhp.schema.oaf.OtherResearchProduct.class));
+        Dataset<eu.dnetlib.dhp.schema.oaf.Publication> sourcePublicationDS = spark.createDataset(
+                sourcePublication.rdd(), Encoders.bean(eu.dnetlib.dhp.schema.oaf.Publication.class));
+        Dataset<eu.dnetlib.dhp.schema.oaf.Software> sourceSoftwareDS = spark.createDataset(
+                sourceSoftware.rdd(), Encoders.bean(eu.dnetlib.dhp.schema.oaf.Software.class));
+
+        Column idIsNotDedup = not(col("id").like("%dedup%"));
+        Column oidIsResultType = col("oid").like(InfoSpaceConstants.ROW_PREFIX_RESULT + "%");
+
+        Dataset<Row> resultIdMapDF = sourceDatasetDS.select(col("id"), explode(col("originalId")).as("oid"))
+                .union(sourceOtherResearchProductDS.select(col("id"), explode(col("originalId")).as("oid")))
+                .union(sourcePublicationDS.select(col("id"), explode(col("originalId")).as("oid")))
+                .union(sourceSoftwareDS.select(col("id"), explode(col("originalId")).as("oid")))
+                .where(idIsNotDedup)
+                .where(oidIsResultType)
```

Comment:
Let's discuss the necessity of this filtering (already mentioned in PR comments).

Reply:
OK, we need both of them.